holidays/countries/sri_lanka.py

```diff
@@ -0,0 +1,287 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.constants import BANK, PUBLIC
+from holidays.groups import (
+    InternationalHolidays,
+    IslamicHolidays,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    BuddhistCalendarHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class SriLanka(
+    HolidayBase,
+    InternationalHolidays,
+    IslamicHolidays,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    BuddhistCalendarHolidays,
+):
+    """
+    https://en.wikipedia.org/wiki/Culture_of_Sri_Lanka#List_of_holidays
+    https://www.timeanddate.com/holidays/sri-lanka/
+    https://factsanddetails.com/south-asia/Srilanka/People_Srilanka/entry-7978.html
+    https://help.wendywutours.com.au/knowledge/public-holidays-and-festivals-sri-lanka
+
+    """
+
+    country = "LK"
+    supported_categories = (BANK, PUBLIC)
+    default_language = "si_LK"
+    # %s (estimated).
+    estimated_label = tr("%s (අනුමානිත)")
+    supported_languages = ("en_US", "si_LK", "ta_LK")
+    # Sri Lanka's regaining of full independence from British influence.
+    start_year = 1948
+
+    def __init__(self, *args, **kwargs):
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, SriLankaIslamicHolidays)
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self)
+        BuddhistCalendarHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Independence Day.
+        self._add_holiday_feb_4(tr("ස්වාධීනත්ව දිනය"))
+
+        if self._year >= 1949:
+            # Thai Pongal.
+            self._add_holiday_jan_14(tr("தைப்பொங்கல்"))
+
+        # New Year.
+        self._add_new_years_day(tr("නව වසර"))
+
+        # Deepavali.
+        self._add_diwali(tr("දීපාවලී"))
+
+        magh_dates = {
+            2022: (FEB, 16),
+            2023: (MAR, 6),
+            2024: (FEB, 24),
+            2025: (FEB, 12),
+            2026: (MAR, 3),
+        }
```

Comment:
Other static holidays should goes to `_CustomBuddhistHoliday` and ` _CustomHinduHolidays` like the Islamic ones as well - see Singapore code for example